vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -716,23 +716,59 @@ import { ItemCache } from '@vaadin/vaadin-grid/src/vaadin-grid-data-provider-mix
         if (index % grid.pageSize != 0) {
           throw 'Got cleared data for index ' + index + ' which is not aligned with the page size of ' + grid.pageSize;
         }
+        // Generates an array of integers from 0 to size-1
+        const generateRange = size => [...Array(size).keys()];
 
-        let firstPage = Math.floor(index / grid.pageSize);
-        let updatedPageCount = Math.ceil(length / grid.pageSize);
+        function findDescendants(pkey,pages) {
+          const parentCache = cache[pkey];
 
-        for (let i = 0; i < updatedPageCount; i++) {
-          let page = firstPage + i;
-          let items = cache[pkey][page];
-          grid.$connector.doDeselection(items.filter(item => selectedKeys[item.key]));
+          if(!parentCache) return [];
+
+          pages = pages ? pages:generateRange(parentCache.length);
+
+          return pages.flatMap(currentPage => {
```

Comment:
Note: this would need to be refactored when cherry-picking to Vaadin 14 as IE11 does not support `flatMap`.

Reply:
Replaced with `map` and `reduce`